examples/dapperish.go

```diff
@@ -19,10 +18,10 @@ import (
 func client() {
 	reader := bufio.NewReader(os.Stdin)
 	for {
-		span := opentracing.StartTrace("getInput")
-		ctx := opentracing.BackgroundGoContextWithSpan(span)
+		ctx, span := opentracing.BackgroundContextWithSpan(
+			opentracing.StartTrace("getInput"))
 		// Make sure that global trace tag propagation works.
-		span.TraceContext().SetTraceAttribute("User", os.Getenv("USER"))
+		span.SetTraceAttribute("User", os.Getenv("USER"))
```

Comment:
I, too, am unnerved by the naming around "Trace Attributes"... if we pursue the basic model of this PR, something like `SetCascadedTag` is worth considering seriously (since both the cascaded and "non-cascaded" tags apply to the same object now: the Span).

@adriancole @michaelsembwever @yurishkuro thoughts on something like the following three methods on Span?

```
    SetTag(string, BasicType)
    SetCascadingTag(string, BasicType)
    Tag(string) --> string
```

Some problems right off the bat:
- If "cascading tags" (trace attributes) are to be used as HTTP headers without bizarro escaping, we run headlong into the lengthy/complex caveats about casing, hyphens, etc. And it seems like the key restrictions for "plain" tags and cascading tags should be identical.
- What to do if there's a "cascading" tag and a user calls `SetTag` with the same key? Who wins?